hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -439,6 +468,11 @@ public DatanodeDeletedBlockTransactions getTransactions(
           keyValue = iter.next();
           DeletedBlocksTransaction txn = keyValue.getValue();
           final ContainerID id = ContainerID.valueOf(txn.getContainerID());
+          if (VersionedDatanodeFeatures.isFinalized(HDDSLayoutFeature.DATA_DISTRIBUTION) &&
+              txn.hasTotalBlockReplicatedSize()) {
+            txSizeMap.put(txn.getTxID(),
```

Comment:
We can move this population for transaction getting send to DN, in getTransaction() where transactions is populated, to be removed -- can not find any action for removal

Reply:
As we all aware, block deletion command have all kinds of event, time out, retry commands, duplicate commands, duplicate responses, replicas of one block deleted on some DNs, but not the others. So the design of SCM side deletion transaction summary, since it's persisted, first consideration is accuracy, robust, we don't want to see a negative summary due to corner cases not covered,  nor do we want to add more complexity to current block deletion flow. As the summary acts like a global metrics for all three SCMs, so it's chosen to use the DB content as the source of summary, that's whey it's in DeletedBlockLogImpl, not the SCMDeletedBlockTransactionStatusManager.